lua/fidget.lua

```diff
@@ -316,7 +316,14 @@ function base_fidget:close()
     self.winid = nil
   end
   if self.bufid ~= nil and api.nvim_buf_is_valid(self.bufid) then
-    api.nvim_buf_delete(self.bufid, { force = true })
+    -- If the fidget buffer becomes the current one after closing the fidget
+    -- window, this most likely means that vim is exiting, although VimLeavePre
+    -- autocmd was not executed yet.
+    -- Deleting the buffer at this point causes a Neovim crash.
+    -- We let the buffer be - it will be deleted automatically on vim exit.
+    if self.bufid ~= api.nvim_get_current_buf() then
```

Comment:
This workaround seems to no longer be necessary with https://github.com/neovim/neovim/issues/20197 fixed. I was able to do the crash reproduction instructions from the PR description without Neovim crashing.

I'm just letting you know in case you want to remove this workaround for nightly. The workaround is nott harmful, so there is no rush removing it and it may be still useful for people not running nightly.

Reply:
Let's leave it in for now. I don't even personally use nightly on all of my machines and I'd like this plugin to work without having to compile the latest (: